test/fixtures/compound12.css

```diff
@@ -0,0 +1,18 @@
+
+#foo a b + c d li {
+	--color: #ffffff;
+	color: var(--color);
+}
+
+#foo.bar a c b + c d e {
+	--color: #000000;
+}
+
+#bar a b + c d li {
+	--color2: #ffffff;
+	color: var(--color2);
+}
+
+#bar.bar a c b c + d e {
```

Comment:
I assume this isn't in the output because `#bar.bar a c b c` does not match `#bar a b` and the sibling is off as well.

Let's add a comment above this case about why it does not match and we should not see it in the output.